src/test/documentation_examples.rs

```diff
@@ -1414,9 +1422,74 @@ async fn versioned_api_examples() -> Result<()> {
     let client = Client::with_options(options)?;
     // End Versioned API Example 4
 
+    let mut options = ClientOptions::parse(&uri).await?;
+    let server_api = ServerApi::builder()
+        .version(ServerApiVersion::V1)
+        .strict(true)
+        .build();
+    options.server_api = Some(server_api);
+    let client = Client::with_options(options)?;
+    let db = client.database("versioned-api-migration-examples");
+    db.collection::<Document>("sales").drop(None).await?;
+    // Start Versioned API Example 5
+    db.collection("sales").insert_many(vec![
+        doc! { "_id" : 1, "item" : "abc", "price" : 10, "quantity" : 2, "date" : iso_date("2021-01-01T08:00:00Z")? },
+        doc! { "_id" : 3, "item" : "xyz", "price" : 5, "quantity" : 5, "date" : iso_date("2021-02-03T09:05:00Z")? },
+        doc! { "_id" : 2, "item" : "jkl", "price" : 20, "quantity" : 1, "date" : iso_date("2021-02-03T09:00:00Z")? },
+        doc! { "_id" : 4, "item" : "abc", "price" : 10, "quantity" : 10, "date" : iso_date("2021-02-15T08:00:00Z")? },
+        doc! { "_id" : 5, "item" : "xyz", "price" : 5, "quantity" : 10, "date" : iso_date("2021-02-15T09:05:00Z")? },
+        doc! { "_id" : 6, "item" : "xyz", "price" : 5, "quantity" : 5, "date" : iso_date("2021-02-15T12:05:10Z")? },
+        doc! { "_id" : 7, "item" : "xyz", "price" : 5, "quantity" : 10, "date" : iso_date("2021-02-15T14:12:12Z")? },
+        doc! { "_id" : 8, "item" : "abc", "price" : 10, "quantity" : 5, "date" : iso_date("2021-03-16T20:20:13Z")? }
+    ], None).await?;
+    // End Versioned API Example 5
+
+    let result = db
+        .run_command(
+            doc! {
+                "count": "sales"
+            },
+            None,
+        )
+        .await;
+    // Start Versioned API Example 6
+    let err = CommandError {
+        message: "Provided apiStrict:true, but the command count is not in API Version 1. Information on supported commands and migrations in API Version 1 can be found at https://dochub.mongodb.org/core/manual-versioned-api".to_string(),
+        code: 323,
+        code_name: "APIStrictError".to_string(),
+    };
+    // End Versioned API Example 6
+    let got_err = if let ErrorKind::Command(ref e) = *result.as_ref().unwrap_err().kind {
+        e
+    } else {
+        panic!("invalid result {:?}", result);
+    };
+    assert_eq!(err, *got_err);
```

Comment:
rather than using eq here (which impacts the public API), I think it's sufficient to compare code and codename.

Reply:
Done.